scripts/ai/controllers/PipeController.lua

```diff
@@ -285,12 +286,27 @@ function PipeController:measurePipeProperties()
 end
 
 --- Unfolds the pipe completely to measure the pipe properties.
-function PipeController:instantUnfold()
-    Foldable.setAnimTime(self.implement, 0, true)
-    if self.pipeSpec.hasMovablePipe then
-        self.implement:setPipeState(PipeController.PIPE_STATE_OPEN, true)
-        self.implement:updatePipeNodes(999999, nil)
-        self.implement:setAnimationTime(self.pipeSpec.animation.name, 1, true, false)
+function PipeController:instantUnfold(ignorePipeState)
+    if self.pipeSpec.currentState ~= self.pipeSpec.targetState or 
+        self.pipeSpec.currentState ~= PipeController.PIPE_STATE_OPEN or ignorePipeState then
+        if self.foldableSpec.turnOnFoldDirection == -1 then
+            Foldable.setAnimTime(self.implement, 0, true)
+        else 
+            Foldable.setAnimTime(self.implement, 1, true)
+        end
+        if self.pipeSpec.hasMovablePipe then
```

Comment:
We should really explain why are we doing this, open/close/open, plus some animation magic is far from self explanatory, will help us later to remember what this hack is doing.

Reply:
Of cause you are right, but I don't really now how I made this working in first place. Mostly alot of try and error and praying.